Can pre-release infra be used to build MOM6 standalone

chrisb13 opened the pull request:
Attempting to run mom6 standalone, Harshula said we can use pre-release infra'!

---
:rocket: The latest prerelease `access-om3/pr151-4` at 279fb07b9836feaf87d312443adea72c36f9e561 is here: https://github.com/ACCESS-NRI/ACCESS-OM3/pull/151#issuecomment-3342432820 :rocket:

chrisb13 committed:
Update spack.yaml

Attempting to run mom6 standalone, Harshula said we can use pre-release infra'!

chrisb13 commented:
ping @harshula as we've been chatting about it.

chrisb13 commented:
Closing this as @dougiesquire has [re-confirmed that it's not possible](https://access-nri.zulipchat.com/#narrow/channel/470332-ocean-seaice/topic/Pipeline.20for.20MOM6.20node.20testing/near/541140203).

dougiesquire commented:
It is possible, but what you tried here is not how to do it.

As we talked about yesterday, you can use the prerelease infrastructure to build whatever you like (within the confines of a few CI checks/requirements). E.g. in [this PR](https://github.com/ACCESS-NRI/ACCESS-OM3/pull/45) I built FRE-NCtools. However, this is not something we should do routinely or probably at all.

chrisb13 committed:
Update spack.yaml

Trying out a variation of what Harshula suggested here: https://access-nri.zulipchat.com/#narrow/channel/470332-ocean-seaice/topic/Pipeline.20for.20MOM6.20node.20testing/near/541139569
(Dougie suggested this might work without deleting the other components...)

chrisb13 commented:
From @CodeGat [the CD wizard](https://access-nri.zulipchat.com/#narrow/channel/470332-ocean-seaice/topic/Pipeline.20for.20MOM6.20node.20testing/near/541948675):
>Haha, the hack works, it's the comment that is wrong. You'd use the module access-mom6/pr151-4 instead. Undocumented, but hey, it's a hack :wink:


```bash
[cyb561@gadi-login-04 ~]$ module purge
[cyb561@gadi-login-04 ~]$ module use /g/data/vk83/prerelease/modules
[cyb561@gadi-login-04 ~]$ module load access-mom6/pr151-4
Loading access-mom6/pr151-4
  Loading requirement: access-mom6/dependencies/pr151-4/access-mocsy/2025.07.002-ucihukj access-mom6/dependencies/pr151-4/access-generic-tracers/2025.08.000-lbeknxx
[cyb561@gadi-login-04 ~]$ which access-mom6
/usr/bin/which: no access-mom6 in (/g/data/vk83/prerelease/apps/spack/0.22/release/linux-rocky8-x86_64_v4/oneapi-2025.2.0/access-mom6-2025.07.000-ruhunvj5oyc2nidysvbmajb42ehtszzm/bin:/home/561/cyb561/.local/bin:/home/561/cyb561/bin:/opt/pbs/default/bin:/opt/nci/bin:/opt/bin:/opt/Modules/v4.3.0/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
```

aidanheerdegen commented:
Using the wrong exe name

```
$ ls /g/data/vk83/prerelease/apps/spack/0.22/release/linux-rocky8-x86_64_v4/oneapi-2025.2.0/access-mom6-2025.07.000-ruhunvj5oyc2nidysvbmajb42ehtszzm/bin
mom6
```

chrisb13 commented:
Thanks @aidanheerdegen, I was hoping to have a hacked `config.yaml` with something like:
```yaml
modules:
    use:
        - /g/data/vk83/prerelease/modules
    load:
        - access-mom6/pr151-4

model: access-mom6
exe: mom6
input: 
.
.
```
Would you expect this to work?

I suppose another option is:
```yaml
model: mom6
exe: $ /g/data/vk83/prerelease/apps/spack/0.22/release/linux-rocky8-x86_64_v4/oneapi-2025.2.0/access-mom6-2025.07.000-ruhunvj5oyc2nidysvbmajb42ehtszzm/bin
mom6/mom6
```
Which I think should work for sure but is less preferred (by the release team I think)?

dougiesquire commented:
Both the `model` and `exe` should be `mom6`

chrisb13 commented:
Thanks @dougiesquire, like this?

```yaml
modules:
    use:
        - /g/data/vk83/prerelease/modules
    load:
        - access-mom6/pr151-4

model: mom6
exe: mom6
```

dougiesquire commented:
Yup. The `model` field is the name of the [Payu model driver](https://github.com/payu-org/payu/blob/33045e577eea6e74cfcb957c9f7a255c4369b6d2/payu/models/__init__.py#L24-L49) to use. The `exe` field is the path to the executable (or just the executable name if it is already on your `PATH`)

dougiesquire commented:
@chrisb13 I think this can be closed?

chrisb13 commented:
[Yes](https://github.com/ACCESS-NRI/ACCESS-OM3/pull/218#issuecomment-4203325208), thanks for your patience.
